lib/galaxy/tools/parameters/basic.py

```diff
@@ -235,6 +234,17 @@ def build( cls, tool, param ):
         else:
             return parameter_types[param_type]( tool, param )
 
+    @classmethod
+    def parse_name(cls, input_source):
+        name = input_source.get("name", None)
+        if name is None:
+            argument = input_source.get("argument")
+            if argument:
+                name = argument.lstrip("-")
+            else:
```

Comment:
All existing tools would have parameters that are then annotated as having arguments that may have nothing to do with the underlying application or wrapper then.